<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PageStepperMotorWgtClass</class>
 <widget class="QWidget" name="PageStepperMotorWgtClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>386</width>
    <height>986</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,0,0">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>9</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Motor parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="bottomMargin">
       <number>9</number>
      </property>
      <property name="horizontalSpacing">
       <number>24</number>
      </property>
      <item row="6" column="0">
       <widget class="QCheckBox" name="limitSpeedChk">
        <property name="text">
         <string>Limit speed with max speed</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QCheckBox" name="useMaxSpeedChk">
        <property name="text">
         <string>Move with max speed</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QCheckBox" name="currentAsRMSchk">
        <property name="text">
         <string>Current as RMS</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QSpinBox" name="ratedCurrBox">
        <property name="suffix">
         <string> mA</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="reversChk">
        <property name="text">
         <string>Revers</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="nominalCurrentLbl">
        <property name="text">
         <string>     Nominal current</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="maxNominalSpeedLbl">
        <property name="text">
         <string>     Max nominal speed</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="StepSpinBox" name="maxSpeedBox" native="true">
        <property name="enabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Motion setup</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3" columnstretch="0,0">
      <property name="horizontalSpacing">
       <number>12</number>
      </property>
      <item row="5" column="0">
       <widget class="QCheckBox" name="accelChk">
        <property name="text">
         <string>Acceleration</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="StepSpinBox" name="antiplaySpeedBox" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="StepSpinBox" name="speedBox" native="true"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="antiplaySpeedLbl">
        <property name="text">
         <string>     Backlash compensation speed</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="decelLbl">
        <property name="text">
         <string>     Deceleration</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="playChk">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Backlash compensation</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="StepSpinBox" name="accelBox" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="suffix" stdset="0">
         <string> steps/s</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="StepSpinBox" name="decelBox" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="suffix" stdset="0">
         <string> steps/s</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="StepSpinBox" name="playBox" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="suffix" stdset="0">
         <string> steps</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="workingSpeedLbl">
        <property name="text">
         <string>     Working speed</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="stepsPerTurnLbl">
        <property name="text">
         <string>Steps per turn</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QSpinBox" name="stepsPerTurnBox">
        <property name="suffix">
         <string> steps</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="2">
       <widget class="QFrame" name="frame_microstep">
        <property name="frameShape">
         <enum>QFrame::Box</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="lineWidth">
         <number>0</number>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>7</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Microstep mode</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="stepmodeBox">
           <item>
            <property name="text">
             <string>full step</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1/2 step</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1/4 step</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1/8 step</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1/16 step</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1/32 step</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1/64 step</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1/128 step</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1/256 step</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="feedbackBox">
     <property name="title">
      <string>Feedback</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4" columnstretch="0,0">
      <property name="horizontalSpacing">
       <number>12</number>
      </property>
      <item row="1" column="0">
       <widget class="QRadioButton" name="feedbackEmfRadio">
        <property name="toolTip">
         <string>Does not use the position sensor (encoder). Provides silent operation.</string>
        </property>
        <property name="text">
         <string>EMF</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QFrame" name="encoderFrame">
        <layout class="QGridLayout" name="gridLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="cptLbl">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Encoder counts per turn</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="encoderReverseChk">
           <property name="text">
            <string>Encoder reverse</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="cptBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="suffix">
            <string> counts</string>
           </property>
           <property name="maximum">
            <number>2147483647</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QRadioButton" name="feedbackNoneRadio">
        <property name="toolTip">
         <string>This algorithm does not have special features.</string>
        </property>
        <property name="text">
         <string>None</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="encoderRadio">
        <property name="toolTip">
         <string>Requires an encoder. Controlled by the PID regulator, with the coordinate as the controlled parameter.</string>
        </property>
        <property name="text">
         <string>Encoder</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QRadioButton" name="encoderMediatedRadio">
        <property name="toolTip">
         <string>This mode of operation is optimally used in systems with large backlash to get to the desired coordinate in a few iterations, on average from 3 to 10.</string>
        </property>
        <property name="text">
         <string>Encoder mediated</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="feedbackTypeLabel">
        <property name="text">
         <string>  Type</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="feedbackTypeBox">
        <item>
         <property name="text">
          <string>Autodetect</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Single-ended</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Differential</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QGroupBox" name="backEMFestimator">
     <property name="title">
      <string>BackEMF estimator</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5" columnstretch="0,0">
      <property name="horizontalSpacing">
       <number>12</number>
      </property>
      <item row="6" column="0" colspan="2">
       <widget class="QFrame" name="inductanceFrame">
        <layout class="QGridLayout" name="gridLayout_inds">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QCheckBox" name="inductanceChk">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Autodetect            </string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="FloatSpinBox" name="cptBox_inds" native="true">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="suffix" stdset="0">
            <string> H</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Inductance</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="QFrame" name="resistanceFrame">
        <layout class="QGridLayout" name="gridLayout_res">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="1" column="0">
          <widget class="QCheckBox" name="resistanceChk">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Autodetect            </string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="FloatSpinBox" name="cptBox_res" native="true">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="suffix" stdset="0">
            <string> Ohm</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Resistance</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="8" column="0" colspan="2">
       <widget class="QFrame" name="KmFrame">
        <layout class="QGridLayout" name="gridLayout_Km">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="1" column="0">
          <widget class="QCheckBox" name="KmChk">
           <property name="text">
            <string>Autodetect</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="FloatSpinBox" name="cptBox_Km" native="true">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="suffix" stdset="0">
            <string> V/s</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Back-EMF coefficient</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>StepSpinBox</class>
   <extends>QWidget</extends>
   <header>stepspinbox.h</header>
  </customwidget>
  <customwidget>
   <class>FloatSpinBox</class>
   <extends>QWidget</extends>
   <header>floatspinbox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>reversChk</tabstop>
  <tabstop>useMaxSpeedChk</tabstop>
  <tabstop>limitSpeedChk</tabstop>
  <tabstop>ratedCurrBox</tabstop>
  <tabstop>playChk</tabstop>
  <tabstop>playBox</tabstop>
  <tabstop>accelChk</tabstop>
  <tabstop>accelBox</tabstop>
  <tabstop>decelBox</tabstop>
  <tabstop>stepsPerTurnBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>playChk</sender>
   <signal>toggled(bool)</signal>
   <receiver>playBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>264</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>accelChk</sender>
   <signal>toggled(bool)</signal>
   <receiver>accelBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>accelChk</sender>
   <signal>toggled(bool)</signal>
   <receiver>decelBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>152</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playChk</sender>
   <signal>toggled(bool)</signal>
   <receiver>antiplaySpeedBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>134</x>
     <y>264</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>useMaxSpeedChk</sender>
   <signal>toggled(bool)</signal>
   <receiver>speedBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inductanceChk</sender>
   <signal>toggled(bool)</signal>
   <receiver>cptBox_inds</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resistanceChk</sender>
   <signal>toggled(bool)</signal>
   <receiver>cptBox_res</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>KmChk</sender>
   <signal>toggled(bool)</signal>
   <receiver>cptBox_Km</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>feedbackEmfRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>backEMFestimator</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>feedbackNoneRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>backEMFestimator</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>encoderRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>backEMFestimator</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>encoderMediatedRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>backEMFestimator</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
